react-state/my-app/src/ToggleButton.js

```diff
@@ -0,0 +1,30 @@
+import { useState } from 'react';
+
+export default function ToggleButton({text, color}) {
+  const [isClicked, setIsClicked] = useState(false);
+
+  console.log('useState:', isClicked); // false on mount; true on re-render
+  function handleClick() {
+    console.log('before setter:', isClicked); // false
+    setIsClicked(!isClicked);
+    console.log('after setter:', isClicked); // false because the data referencing the state hasn't updated.
+  }
+  // either returns on line 14 or 22 re-renders the ToggleButton component, re-running the code.
+  if (!isClicked) {
```

Comment:
This works, but it duplicates a lot of code. See if you can figure out how to avoid the duplication.

Reply:
<img width="912" alt="Screenshot 2023-04-01 at 12 18 06 PM" src="https://user-images.githubusercontent.com/110208589/229310246-7e60078c-5a7f-401b-bb92-2f44f9ae8b38.png">

Reply:
Removed the duplicated code and combined the styles using a ternary.